beluga_ros/include/beluga_ros/amcl.hpp

```diff
@@ -45,33 +58,32 @@ struct AmclParams {
 };
 
 /// Implementation of the Adaptive Monte Carlo Localization (AMCL) algorithm.
+/**
+ * See Probabilistic Robotics \cite thrun2005probabilistic Chapter 6.2.
+ *
+ * \tparam SensorModel Type representing a sensor model. Must satisfy \ref SensorModelPage.
+ **/
+template <typename SensorModel>
 class Amcl {
  public:
   using particle_type = std::tuple<Sophus::SE2d, beluga::Weight>;
 
-  using motion_model_variant = std::variant<
-      beluga::DifferentialDriveModel,     //
-      beluga::OmnidirectionalDriveModel,  //
-      beluga::StationaryModel>;
+  using sensor_model_type = SensorModel;
 
-  using sensor_model_variant = std::variant<
-      beluga::LikelihoodFieldModel<beluga_ros::OccupancyGrid>,  //
-      beluga::BeamSensorModel<beluga_ros::OccupancyGrid>>;
-
-  using execution_policy_variant = std::variant<std::execution::sequenced_policy, std::execution::parallel_policy>;
+  using map_type = typename sensor_model_type::map_type;
```

Comment:
@serraramiro1 meta: do this two need to be coupled like this? I mean, there will be a constraint between the two, but do we have to derive it like this instead of allowing the user to set both?